portal/views/user.py

```diff
@@ -170,15 +170,18 @@ def set_user_groups(user_id):
       - in: body
         name: body
         schema:
-          id: groups
-          required:
-            - name
+          id: nested_groups
           properties:
-            name:
-              type: string
-              description:
-                The string defining the name of each group the user should
-                belong to.  Must exist as an available group in the system.
+            groups:
+              type: array
+              items:
+                type: object
```

Comment:
You may be able to reference the `group_definition` definition/schema here instead of redefining it (ie `$ref: "#/definitions/group_definition"`). The only reason I didn't is the `group_definition` schema requires the `description` attribute while the example you gave me does not.